### ISSUE TITLE
E/StickerPackDetails: Validation failed:pack is marked as animated pack but contains non animated stickers. All stickers in animated pack should be animated sticker.

### ISSUE THREAD

**Issue by hamza-ahmed** [WhatsApp/stickers#763]
E/StickerPackDetails: Validation failed:pack is marked as animated pack but contains non animated stickers. All stickers in animated pack should be animated sticker.

**Comment by MarvinStelter** [comment]
You cant use both in one pack! Convert static webp's in to animated sticker

**Comment by hamza-ahmed** [comment]
All used sticker is animated , but this message showing at me !!

**Comment by MarvinStelter** [comment]
Please send a webp file, I'll look into it

**Comment by hamza-ahmed** [comment]
> All used sticker is animated , but this message showing at me 


Its same assets file using at this code , but im using it from api

**Comment by MarvinStelter** [comment]
Then I guess something went wron while downloading it

**Comment by hamza-ahmed** [comment]
im using this two funcation to convert api stickers to Webp

   public static Uri convertImageToWebP(Uri uri, String StickerBookId, String StickerId, Context context) {
        try {

            Bitmap bitmap = MediaStore.Images.Media.getBitmap(context.getContentResolver(), uri);
            dirChecker(context.getFilesDir() + "/" + StickerBookId);
            String path = context.getFilesDir() + "/" + StickerBookId + "/" + StickerBookId + "-" + StickerId + ".webp";
            Log.w("Conversion Data: ", "path: " + path);
            makeSmallestBitmapCompatible(path, bitmap);
            return Uri.fromFile(new File(path));
        } catch (Exception e) {
            e.printStackTrace();
        }
        --------------------------------------------------------------------------

  private static void makeSmallestBitmapCompatible(String path, Bitmap bitmap) {
        int quality = 100;
        FileOutputStream outs = null;
        try {
            outs = new FileOutputStream(path);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        bitmap = Bitmap.createScaledBitmap(bitmap, 512, 512, true);

        int byteArrayLength = 100000;
        ByteArrayOutputStream bos = null;

        while ((byteArrayLength / 1000) >= 100) {
            bos = new ByteArrayOutputStream();

            bitmap.compress(Bitmap.CompressFormat.WEBP,
                    quality,
                    bos);

            byteArrayLength = bos.toByteArray().length;
            quality -= 10;

            Log.w("IMAGE SIZE IS NOW", byteArrayLength + "");
        }
        try {
            outs.write(bos.toByteArray());
            outs.close();
        } catch (IOException e) {
            e.printStackTrace();
        }

    }

**Comment by MarvinStelter** [comment]
That's wrong you can't save a animated sticker as normal bitmap... Use mobile ffmpeg and codec webp_anim

**Comment by hamza-ahmed** [comment]
oh can help me about that ?

**Comment by MarvinStelter** [comment]
I'm not good at it, I created a sticker maker app and implemented it yesterday.

I can send you a code snippet that converts static images into a gif and that gif will be converted to a animated webp with mobile ffmpeg.

**Comment by hamza-ahmed** [comment]
ok send me that please

**Comment by MarvinStelter** [comment]
https://gist.github.com/MarvinStelter/492cc15ebcf0986c2f8fb3e5cef7b3c7

**Comment by MarvinStelter** [comment]
Not explained but I guess you'll understand it :)

**Comment by hamza-ahmed** [comment]
yes , so im searching about encode apis url to webp

**Comment by hamza-ahmed** [comment]
im tired im cant convert webp urls to anamation webp !!!!1

**Comment by MarvinStelter** [comment]
You shouldn't edit them, make them "ready to add" before downloading not afterwards then it should work if you download them with DownloadManager

**Comment by hamza-ahmed** [comment]
Can we talk with skype ?

**Comment by hamza-ahmed** [comment]
thanks alot its work %%
